From fe09c4cf6b551e970c8251a190fbaa476b958fb2 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Mon, 23 Jun 2014 21:51:16 -0400
Subject: [fs] nfsd: estimate sequence response size

Message-id: <1403560314-32525-135-git-send-email-bfields@redhat.com>
Patchwork-id: 84492
O-Subject: [Patch RHEL7 134/172] nfsd4: estimate sequence response size
Bugzilla: 1111719
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

Otherwise a following patch would turn off all 4.1 zero-copy reads.

Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: ccae70a9ee415a89b70e97a36886ab55191ebaea
Bugzilla: 1111719
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4proc.c b/fs/nfsd/nfs4proc.c
index 7fb0dc5..b666104 100644
--- a/fs/nfsd/nfs4proc.c
+++ b/fs/nfsd/nfs4proc.c
@@ -1564,6 +1564,12 @@ static inline u32 nfsd4_rename_rsize(struct svc_rqst *rqstp, struct nfsd4_op *op
   + op_encode_change_info_maxsz) * sizeof(__be32);
 }
 
+static inline u32 nfsd4_sequence_rsize(struct svc_rqst *rqstp,
+           struct nfsd4_op *op)
+{
+ return NFS4_MAX_SESSIONID_LEN + 20;
+}
+
 static inline u32 nfsd4_setattr_rsize(struct svc_rqst *rqstp, struct nfsd4_op *op)
 {
  return (op_encode_hdr_size + nfs4_fattr_bitmap_maxsz) * sizeof(__be32);
-- 
1.7.1